go/libraries/doltcore/doltdb/commit_itr.go

```diff
@@ -163,114 +162,39 @@ func hashToCommit(ctx context.Context, vrw types.ValueReadWriter, h hash.Hash) (
 	return NewCommit(vrw, cmSt), nil
 }
 
-type CommitIndexingCommitItr struct {
-	ddb         *DoltDB
-	itr         CommitItr
-	commits     []hash.Hash
-	authors     []string
-	commitTimes []time.Time
-}
-
-func NewCommitIndexingCommitItr(ddb *DoltDB, itr CommitItr) *CommitIndexingCommitItr {
-	return &CommitIndexingCommitItr{
-		ddb:         ddb,
-		itr:         itr,
-		commits:     make([]hash.Hash, 0, 4096),
-		authors:     make([]string, 0, 4096),
-		commitTimes: make([]time.Time, 0, 4096),
-	}
-}
-
-func (cmItr *CommitIndexingCommitItr) Reset(ctx context.Context) error {
-	cmItr.commits = make([]hash.Hash, 0, 4096)
-	cmItr.authors = make([]string, 0, 4096)
-	cmItr.commitTimes = make([]time.Time, 0, 4096)
-
-	return cmItr.itr.Reset(ctx)
-}
-
-func (cmItr *CommitIndexingCommitItr) Next(ctx context.Context) (hash.Hash, *Commit, error) {
-	h, cm, err := cmItr.itr.Next(ctx)
-
-	if err != nil {
-		return hash.Hash{}, nil, err
-	}
-
-	meta, err := cm.GetCommitMeta()
-
-	if err != nil {
-		return hash.Hash{}, nil, err
-	}
-
-	commitTS := meta.Time()
-	author := meta.Name
-
-	cmItr.commits = append(cmItr.commits, h)
-	cmItr.commitTimes = append(cmItr.commitTimes, commitTS)
-	cmItr.authors = append(cmItr.authors, author)
-
-	return h, cm, nil
-}
-
-type TimeRange struct {
-	Min time.Time
-	Max time.Time
-}
+// CommitFilter is a function that returns true if a commit should be filtered out, and false if it should be kept
+type CommitFilter func(context.Context, hash.Hash, *Commit) (filterOut bool, err error)
```

Comment:
The logical case is inverted on this. Predicate functions should return true for a match, not true for exclusion. "(accept bool, err error)"